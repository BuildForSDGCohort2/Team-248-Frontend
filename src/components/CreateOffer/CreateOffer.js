import React, { Component } from "react";

class CreateOffer extends Component {
  state = {
    startDate: {
      value: "",
      error: "Invalid, please try again"
    },
    endDate: {
      value: "",
      error: "Invalid, please try again"
    },
    pricePerHour: {
      value: "",
      error: "Invalid, please try again"
    },
    address: {
      value: "",
      error: "Invalid, please try again"
    },
    qualifications: {
      value: "",
      error: "Invalid, please try again"
    },
    snackbar: {
      open: false,
      vertical: "top",
      horizontal: "center"
    }
  };
  render() {
    return <h1>Hello Create Offer</h1>;
  }
}

export default CreateOffer;
