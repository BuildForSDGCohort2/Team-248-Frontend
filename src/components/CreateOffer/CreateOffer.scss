@import "../../assets/css/bootstrap.min.css";
@import "../../assets/scss/paper-kit";

.content{
  width: 30%;
  margin-top: 150px !important;

  .MuiFormControl-root{
    width: 100%;
  }

  h2{
    margin-bottom: 25px;
  }

  button[type="submit"]{
    margin-top: 20px;
  }

  .category-input{
    height: 50px;
    border-color: #bfbdbd;
    background: none;
    border-radius: 5px;
    color: #4a4a4a;
    font-size: medium;
  }
}

.navbar.navbar-transparent {
  padding-top: 0;
  background-color: white !important;
  box-shadow: 0 6px 10px -4px rgba(0, 0, 0, 0.15);
}