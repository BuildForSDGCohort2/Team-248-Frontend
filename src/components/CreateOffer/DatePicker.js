import React from "react";
import { Box } from "@material-ui/core";
import {
  MuiPickersUtilsProvider,
  // KeyboardTimePicker,
  KeyboardDatePicker
} from "@material-ui/pickers";
import DateFnsUtils from "@date-io/date-fns";
const DatePicker = ({ label, value, onChange }) => {
  return (
    <Box width="100%" mb={2}>
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <KeyboardDatePicker
          id="data-picker-dialog"
          label={label}
          inputVariant="outlined"
          format="MM/dd/yyyy"
          value={value}
          onChange={onChange}
          KeyboardButtonProps={{
            "aria-label": "change date"
          }}
        />
      </MuiPickersUtilsProvider>
    </Box>
  );
};

export default DatePicker;
