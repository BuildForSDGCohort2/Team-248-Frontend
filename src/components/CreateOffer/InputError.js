import React from "react";
import { ErrorMessage } from "formik";
const InputError = ({ name, message }) => {
  {
    return (
      <ErrorMessage
        name={name}
        render={() => <div style={{ color: "#FF0000" }}>{message}</div>}
      />
    );
  }
};

export default InputError;
