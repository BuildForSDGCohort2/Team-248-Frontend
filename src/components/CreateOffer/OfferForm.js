import React from "react";
import { Formik, Form, Field, ErrorMessage } from "formik";
import { number, object, string, date } from "yup";
import DatePicker from "./DatePicker";
import { Button, Box, TextField, FormGroup } from "@material-ui/core";

const min = new Date();
const minDate = `${min.getMonth() + 1}/${min.getDate()}/${min.getFullYear()}`;
const max = new Date();

const initialValues = {
  startDate: null,
  endDate: null,
  pricePerHour: null,
  address: null,
  qualifications: null
};
const validationSchema = object({
  startDate: date()
    .default(new Date(min))
    .min(min, `start date should be equal or later than ${minDate}`)
    .required("Start date is required"),
  endDate: date()
    .default(new Date(max))
    .min(min, `end date should be equal or later than ${minDate}`)
    .required("End date is required"),
  pricePerHour: number()
    .required("Price per hour is required")
    .min(1, `you can not choose a price less than 1$ per hour`),
  address: string()
    .min(5)
    .required("You address is required"),
  qualifications: string().min(20)
});

const OfferForm = () => {
  return (
    <Formik
      validationSchema={validationSchema}
      initialValues={initialValues}
      onSubmit={({ values }) => console.log(values)}
    >
      {({ values, setFieldValue }) => (
        <Form>
          <FormGroup>
            <Field
              as={DatePicker}
              name="startDate"
              label="Start date"
              value={values.startDate}
              onChange={value => setFieldValue("startDate", value)}
            />
            <ErrorMessage name="startDate" />
          </FormGroup>
          <FormGroup>
            <Field
              as={DatePicker}
              name="endDate"
              label="End date"
              value={values.endDate}
              onChange={value => setFieldValue("endDate", value)}
            />
            <ErrorMessage name="endDate" />
          </FormGroup>
          <Box width="100%" mb={2}>
            <FormGroup>
              <Field
                variant="outlined"
                name="pricePerHour"
                type="number"
                as={TextField}
                label="Price per hour"
              />
              <ErrorMessage name="pricePerHour" />
            </FormGroup>
          </Box>
          <Box width="100%" mb={2} mt={2}>
            <FormGroup>
              <Field
                variant="outlined"
                name="address"
                type="text"
                as={TextField}
                label="Your address"
              />
              <ErrorMessage name="address" />
            </FormGroup>
          </Box>
          <Box width="100%" mb={2} mt={2}>
            <FormGroup>
              <Field
                variant="outlined"
                name="qualifications"
                type="text"
                as={TextField}
                label="prefered qualifications"
              />
              <ErrorMessage name="qualifications" />
            </FormGroup>
          </Box>
          <Box width="100%" my={2}>
            <Button type="submit" variant="contained" color="primary">
              Submit
            </Button>
          </Box>
        </Form>
      )}
    </Formik>
  );
};

export default OfferForm;
