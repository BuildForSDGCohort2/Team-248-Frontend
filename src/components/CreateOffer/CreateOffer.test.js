import React from "react";
import ReactDom from "react-dom";
import CreateOffer from "./CreateOffer";
import { shallow, mount } from "enzyme";

it("snapshot testing", () => {
  const wrapper = shallow(<CreateOffer />);
  expect(wrapper).toMatchSnapshot();
});
