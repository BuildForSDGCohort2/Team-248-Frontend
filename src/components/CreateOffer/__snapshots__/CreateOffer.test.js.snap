// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={6}
  >
    <WithStyles(ForwardRef(Typography))
      variant="h3"
    >
      Create Offer
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(CssBaseline) />
    <CustomSnackbar
      anchorOrigin={
        Object {
          "horizontal": "center",
          "vertical": "top",
        }
      }
      handleClose={[Function]}
      keyProp="topcenter"
      message="Email has been sent Successfully"
      open={false}
    />
    <OfferForm />
    <Styled(MuiBox)
      mt={8}
    >
      <Copyright />
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
